test_scripts/Resumption/InteriorVehicleData/commonResumptionsInteriorVD.lua

```diff
@@ -0,0 +1,435 @@
+m.setHMIAppId = actions.app.setHMIId
+m.isTableEqual = utils.isTableEqual
+m.isTableContains = utils.isTableContains
+m.fail = actions.run.fail
+m.getActualModuleIVData = rc.state.getActualModuleIVData
+
+--[[ General configuration ]]
+local state = rc.state.buildDefaultActualModuleState(rc.predefined.getRcCapabilities())
+rc.state.initActualModuleStateOnHMI(state)
+
+-- [[ Common Functions ]]
+local function boolToTimes(isTrue)
+  if isTrue then
+    return 1
+  end
+  return 0
+end
+
+local function setSubscriptionModuleStatus(pModuleType, pModuleId, isSubscribed)
+  local newValue = {
+    moduleType = pModuleType,
+    moduleId = pModuleId
+  }
+
+  if #modulesWithSubscription == 0 and isSubscribed == true then
+    table.insert(modulesWithSubscription, newValue)
+    return
+  end
+
+  for key, value in pairs(modulesWithSubscription) do
+    if m.isTableEqual(value, newValue) and isSubscribed == false then
+      table.remove(modulesWithSubscription, key)
+      return
+    elseif m.isTableEqual(value, newValue) and isSubscribed == true then
+      return
+    elseif key == #modulesWithSubscription and isSubscribed == true then
+      table.insert(modulesWithSubscription, newValue)
+      return
+    end
+  end
+end
+
+function  m.GetInteriorVehicleData(pModuleType, pModuleId, isSubscribe, pIsIVDataCached,
+    hashChangeExpectTimes, pAppId, pResultCode)
+  pAppId = pAppId or 1
+  local rpc = "GetInteriorVehicleData"
+  if pIsIVDataCached == nil then pIsIVDataCached = false end
+
+  local moduleId = pModuleId or m.getModuleId(pModuleType,  1)
+  local mobileRequestParams = m.cloneTable(rc.rpc.getAppRequestParams(rpc, pModuleType, pModuleId, isSubscribe))
+
+  local hmiRequestParams = rc.rpc.getHMIRequestParams(rpc, pModuleType, moduleId, pAppId, isSubscribe)
+  if hashChangeExpectTimes == 0 then hmiRequestParams.subscribe = nil end
+
+  local cid = m.getMobileSession(pAppId):SendRPC(rc.rpc.getAppEventName(rpc), mobileRequestParams)
+  m.getHMIConnection():ExpectRequest(rc.rpc.getHMIEventName(rpc), hmiRequestParams)
+  :Times(boolToTimes(not pIsIVDataCached))
+  :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+        rc.rpc.getHMIResponseParams(rpc, pModuleType, moduleId, isSubscribe))
+    end)
+
+  local resultCode = pResultCode or "SUCCESS"
+  local responseParams = rc.rpc.getAppResponseParams(rpc, true, resultCode, pModuleType, moduleId, isSubscribe)
+  if pResultCode == "WARNINGS" then
+    responseParams["info"] = "App is already subscribed to the provided module"
+  end
+  m.getMobileSession(pAppId):ExpectResponse(cid, responseParams)
+  :Do(function()
+      setSubscriptionModuleStatus(pModuleType, moduleId, isSubscribe)
+    end)
+
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange")
+  :Do(function(_,data)
+      m.hashId[pAppId] = data.payload.hashID
+    end)
+  :Times(hashChangeExpectTimes or 1)
+end
+
+function m.getModuleId(pModuleType, pModuleIdNumber)
```

Comment:
@jacobkeeler This function goal is to return `moduleId` for default (1) or not default (2) module. So `out.moduleId` is `moduleId` for default module and if `pModuleIdNumber == 1` then we return it, in other case we find another module of requested type and return its `moduleId`

Reply:
Is it possible that you could just return them in whatever order they appear in the object? Of course this would only work if the order that the `for` loop is consistent. If not, I would just suggest that you at least include a description which includes this detail

Reply:
@jacobkeeler The order of `for` loop which uses `pairs` is not consistent. 
I have added short description for the function.